src/opera/pge/dist_s1/dist_s1_pge.py

```diff
@@ -271,6 +331,61 @@ def _validate_rtcs(self):
         self.__validate_rtc_homogeneity(current_matches)  # Per ADT: Baseline can be heterogeneous
         self.__validate_rtc_ordering(all_rtcs)
         self.__validate_co_and_cross_polarizations(all_rtcs)
+        self.__validate_no_duplicates(baseline_matches + current_matches)
+
+    def _validate_previous_product(self):
+        """
+        Run validations for previous product input.
+
+        Checks:
+        1. All files exist
+        2. All files are nonempty
+        3. All files are .tif files
+        4. Correct number of files (one for each TIFF layer) provided
+        5. All files are named conformant to DIST-S1 product specification
+        6. Exact set of layers provided (ie, no duplicates)
+        """
+        sas_config = self.runconfig.sas_config
+        previous_product = sas_config["run_config"].get("pre_dist_s1_product", [])
+
+        check_input_list(
+            previous_product,
+            self.logger,
+            self.name,
+            valid_extensions=self._valid_input_extensions,
+            check_zero_size=True
+        )
+
+        geotiff_layer_names = set([layer for layer in _valid_layer_names if layer != 'BROWSE'])
+
+        if len(previous_product) != len(geotiff_layer_names):
+            msg = f'Unexpected number of files in previous product: {len(previous_product)}'
+            self.logger.critical(
+                self.name,
+                ErrorCode.INVALID_INPUT,
+                msg
+            )
+
+        product_band_matches = [_granule_filename_re.match(basename(f)) for f in previous_product]
```

Comment:
Another nitpick, but since we'll be changing `previous_product` to `previous_products`, I would also rather see `previous_product` used here instead of `f`. Best to avoid one-letter variables names as a general rule.